### ISSUE TITLE
Add analyzer that catches properties that can be made readonly

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-analyzers#3470]
**Version Used**: 

**Steps to Reproduce**:

It would be nice to have an analyzer that catches that the following property can be made readonly. Hit this on a [PR](https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/239223?path=%2Fsrc%2FMicrosoft.VisualStudio.ProjectSystem.Implementation%2FBuild%2FCommonBuildPropertiesProvider.cs&_a=overview) just recently.

``` C#

    public class Foo
    {
        public Foo(string property)
        {
            Property = property;
        }
        
        private string Property
        {
            get;
            set;  // <!-- Can remove
        }

        public void Method()
        {
            Console.WriteLine(Property);
        }
    }
```

**Comment by Evangelink** [comment]
I am curious if we need a new property or to simply extend the unused member rule to report on this unused setter. Any preference @mavasani and @davkean ?

**Comment by mavasani** [comment]
I think this should be done in Roslyn, we already have an analyzer which reports unused/unread private members: https://github.com/dotnet/roslyn/blob/master/src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

This analyzer already has this information and we would just need to add small additional logic to it. @jinujoseph can this issue be transferred back to Roslyn? @Evangelink feel free to pick it up and submit a PR to Roslyn if you are interested.
